lib/tdf3/src/tdf.ts

```diff
@@ -926,111 +927,137 @@ async function unwrapKey({
   }
   const { keyAccess } = manifest.encryptionInformation;
   const splitPotentials = splitLookupTableFactory(keyAccess, allowedKases);
-
-  let responseMetadata;
   const isAppIdProvider = authProvider && isAppIdProviderCheck(authProvider);
-  // Get key access information to know the KAS URLS
-  const rewrappedKeys: Uint8Array[] = [];
 
-  for (const [splitId, potentials] of Object.entries(splitPotentials)) {
+  // Process all splits in parallel and take the first successful one
+  const splitPromises = Object.entries(splitPotentials).map(async ([splitId, potentials]) => {
     if (!potentials || !Object.keys(potentials).length) {
       throw new UnsafeUrlError(
         `Unreconstructable key - no valid KAS found for split ${JSON.stringify(splitId)}`,
         ''
       );
     }
 
-    // If we have multiple ways of getting a value, try the 'best' way
-    // or maybe retry across all potential ways? Currently, just tries them all
-    const [keySplitInfo] = Object.values(potentials);
-    const url = `${keySplitInfo.url}/${isAppIdProvider ? '' : 'v2/'}rewrap`;
+    // Try all potential KAS sources for this split
+    const errors: Array<Error | AxiosError> = [];
+    const kasPromises = Object.values(potentials).map(async (keySplitInfo) => {
+      try {
```

Comment:
[Promise.any](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/any) does all this for you, even having a custom type `AggregateError` that is what you expect.

```suggestion
    const kasPromises = Object.values(potentials).map(async (keySplitInfo) => {
```